const Order = require('../models/Order');

exports.addOrder = async (req, res) => {
    try {
        const { userId, products, totalPrice, shippingCost, subtotal, address, email, number, full_name } = req.body;
        const order = new Order({
            userId,
            products,
            totalPrice,
            shippingCost,
            subtotal,
            address,
            email,
            number,
            full_name
        });
        await order.save();
        res.status(201).json({success: true, order});
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
}

exports.getOrders = async (req, res) => {
    try {
        const orders = await Order.find().populate('userId', 'name email');
        res.status(200).json(orders);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
}
exports.getOrderById = async (req, res) => {
    try {
        const order = await Order.findById(req.params.id).populate('userId', 'name email');
        if (!order) {
            return res.status(404).json({ message: 'Order not found' });
        }
        res.status(200).json(order);
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
}
exports.updateOrder = async (req, res) => {
    try {
        const order = await Order.findByIdAndUpdate(req.params.id, req.body, { new: true });
        if (!order) {
            return res.status(404).json({ message: 'Order not found' });
        }   
        res.status(200).json({success: true, order: order});
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
}
exports.deleteOrder = async (req, res) => {
    try {
        const order = await Order.findByIdAndDelete(req.params.id);
        if (!order) {
            return res.status(404).json({ message: 'Order not found' });
        }
        res.status(200).json({ message: 'Order deleted successfully' });
    } catch (error) {
        res.status(500).json({ message: error.message });
    }
}
